tests/unit/parallel/executors/shared_parallel_executor.cpp

```diff
@@ -32,6 +33,17 @@ struct shared_parallel_executor
     {
         return hpx::async(std::forward<F>(f));
     }
+
+    template <typename F>
+    typename std::remove_reference<
+        typename hpx::util::result_of<
+            typename hpx::util::decay<F>::type()
+        >::type
+    >::type
```

Comment:
Shouldn't this be rather done in the trait? Is it an error in general to return a reference from the passed function?